libtransact/src/contract/address/double_key_hash.rs

```diff
@@ -48,6 +52,35 @@ impl Addresser<(String, String)> for DoubleKeyHashAddresser {
     }
 }
 
+// Used to validate the lengths of the input, including the prefix length and optional custom hash
+// length, which are used to construct the DoubleKeyHashAddresser.
+fn validate_lengths(
+    prefix_length: usize,
+    first_length: Option<usize>,
+) -> Result<(), AddresserError> {
+    // Validate the length of the prefix is not greater than the ADDRESS_LENGTH.
+    if prefix_length > ADDRESS_LENGTH {
+        return Err(AddresserError {
+            message: format!("Prefix length is greater than {}", ADDRESS_LENGTH),
```

Comment:
Suggest: ("Prefix length ({}) is greater than total address length ({})", prefix_length, ADDRESS_LENGTH)